### DISCUSSION TITLE
How to load a VST instrument plugin into Cardinal and play?

### DISCUSSION THREAD

**Discussion by tommai78101** [DISTRHO/Cardinal#605] [Q&A] [answered]
I'm using Ildaeil Plugin Host in Cardinal to load a VST instrument plugin, and I wanted to try and route it so that the Host MIDI controls the plugin, and sends the instrument plugin signals to the Host Audio.

I'm not using any other DAWs. I only have a MIDI keyboard that I can send MIDI signals into Cardinal using Host MIDI.

How should I route the VST instrument plugin accordingly in Cardinal? All I need to do is to play the VST instrument plugin. Nothing fancy required.

If Ildaeil Plugin Host is not the correct way to load VST instrument plugins into Cardinal, what guides can I find that will let me do this?

![image](https://github.com/DISTRHO/Cardinal/assets/6444116/3bd5bf30-c4e3-4601-9081-d8183fd84389)

Thanks in advance.

**Comment by tommai78101** [comment]
Ok, I finally did it.

1. Get Carla. (https://github.com/falkTX/Carla)
2. Get Cardinal. (https://github.com/DISTRHO/Cardinal)
3. In Carla, click on "Configure Carla".
4. In the Plugin Paths, add the Cardinal plugins path.
5. Press OK to save the new plugin path changes.
6. In Carla, click on "Add Plugins".
7. Click on "Refresh" to load the new Cardinal plugins into Carla.
8. Add the "Cardinal FX" plugin.
9. In Carla, in the toolbar menu, click on "Canvas" -> "Show External".
10. In the Carla canvas, map any of the "Readable MIDI ports" to "Carla's midi-in".
11. In Carla, in the toolbar menu, click on "Canvas" -> "Show Internal".
12. In Carla canvas, add your VST instrument plugin and CardinalFX plugins into Carla.
13. Connect the MIDI input "capture 1" output node to both plugins' "events-in" input nodes.
14. Connect your VST instrument plugin's "output nodes" to CardinalFX "input nodes".
15. Connect CardinalFX "output nodes" to the Audio Output's "Playback" nodes.
16. In Carla canvas, double-click on the CardinalFX patch node. Carla will then load Cardinal.
17. In CardinalFX, connect Host Audio "Output Nodes" to any module's input nodes. Then at the end of every connection, connect the module's output nodes to the Host Audio's "Input Nodes" to complete the audio pass-through.
18. Done.

**Comment by dromer** [comment, marked as answer]
You need to use the input Midi Expander on the left of Ildaeil so that CV signals are converted to MIDI for the plugin: https://github.com/DISTRHO/Cardinal/blob/main/docs/CARDINAL-MODULES.md#expanders-midi

![](https://i.imgur.com/e0OIXBy.png)

**Reply by tommai78101** [reply]
Ok, I didn't know the MIDI expanders are supposed to be used like that. Thanks for clearing that up.

Does this still need to use Carla? Or this can be done straight up with CardinalNative.exe?

**Reply by dromer** [reply]
Carla and Ildaeil modules should be built into Cardinal.

(these midi expanders work for both)

**Reply by tommai78101** [reply]
Thanks again.
